modules/consensus/consensusset.go

```diff
@@ -141,9 +143,19 @@ func New(gateway modules.Gateway, persistDir string) (*ConsensusSet, error) {
 		// Register RPCs
 		gateway.RegisterRPC("SendBlocks", cs.rpcSendBlocks)
 		gateway.RegisterRPC("RelayBlock", cs.rpcRelayBlock) // COMPATv0.5.1
-		gateway.RegisterRPC("RelayHeader", cs.rpcRelayHeader)
+		gateway.RegisterRPC("RelayHeader", cs.threadedRPCRelayHeader)
 		gateway.RegisterRPC("SendBlk", cs.rpcSendBlk)
 		gateway.RegisterConnectCall("SendBlocks", cs.threadedReceiveBlocks)
+		cs.tg.OnStop(func() {
+			cs.mu.Lock()
+			defer cs.mu.Unlock()
```

Comment:
unregistering won't affect in-progress RPCs. In some cases, an RPC calls another RPC, so there's a "race condition" where RPC1 succeeds, but then RPC2 fails (because the recipient unregistered it). But this should be benign as long as the RPC error is handled correctly.